tools/shared/src/main/scala/org/scalajs/core/tools/linker/backend/emitter/JSGen.scala

```diff
@@ -147,27 +151,42 @@ private[emitter] final class JSGen(val semantics: Semantics,
   }
 
   def genRawJSClassConstructor(className: String)(
-      implicit globalKnowledge: GlobalKnowledge, pos: Position): Tree = {
+      implicit globalKnowledge: GlobalKnowledge,
+      pos: Position): WithGlobals[Tree] = {
```

Comment:
I had understood. ;) Not having the overload with `keepAllGlovalVarNames` defaulting to true is what was tedious.

Not having `genRawJSClassConstructor` at all would be insane.